src/frontend/context/authcontext.jsx

```diff
@@ -0,0 +1,82 @@
+import axios from "axios";
+import React, { createContext, useContext, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Notify } from "../components/Toast";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const navigate = useNavigate();
+
+  const user = JSON.parse(localStorage.getItem("videoUser"));
+  const token = localStorage.getItem("videoToken");
+
+  const [localToken, setLocalToken] = useState(token ?? "");
+  const [localUser, setLocalUser] = useState(user ?? {});
```

Comment:
All the code is well written and organized, just a suggestion you can also put the userdetails + token in one state as they both are related as a object `{user, token}` rather than creating 2 states